exampleRios.py

```diff
@@ -0,0 +1,131 @@
+      else:
+        self.rs = self.pos_changes
+
+    if len(self.prices) >= self.size:
+      rsi = 100 - 100/(1 + self.rs)
+      if rsi >= self.high_limit:
+        orders.append(Order(event.instrument, -1, 0))
+        orders.append(Order(event.instrument, -1, 0))
+        self.buy = False
+      elif rsi <= self.low_limit:
+        orders.append(Order(event.instrument, 1, 0))
+        orders.append(Order(event.instrument, 1, 0))
+        self.buy = True
+      else:
+        if self.buy:
+          orders.append(Order(event.instrument, 1, 0))
+        else:
+          orders.append(Order(event.instrument, -1, 0))
+
+    return orders
+
+class RiosStrat(Strategy):
+  def __init__(self):
+    self.prices = []
+    self.threshold = 500
+    self.acum = 0
+    self.growing = None
+
+  def push(self, event):
+    price = event.price[0]
+    self.prices.append(price)
+    orders = []
+
+    curr = self.prices[len(self.prices)-1]
+    past = self.prices[len(self.prices)-2]
+    if curr > past:
+      if self.acum >= self.threshold or self.acum <= -self.threshold:
+        self.growing = True
+      self.acum += curr - past
+
+    else:
+      if self.acum >= self.threshold or self.acum <= -self.threshold:
+        self.growing = False
+      self.acum -= past - curr
+
+    if self.acum >= self.threshold or self.acum <= -self.threshold:
+      if self.growing == True:
+        self.growing = False
+        orders.append(Order(event.instrument, 1, 0))
+      else:
+        orders.append(Order(event.instrument, -1, 0))
+   
+    return orders
+
+class PBRStrat(Strategy):
+
+  def __init__(self):
+    self.petr3 = None
+    self.spread = 0.1 # 10%
+    self.ti = 0.482
+    self.tc = 0.298
+    self.orders = []
+    self.first = True
+    self.pbr = None
+
+  def operate(self, event):
+    if event.instrument == "PETR3":
+      self.petr3 = event.price[0]
+      
+    elif event.instrument == "USDBRL":
+      self.pbr = (self.petr3 / event.price[0]) * self.ti + self.tc
+
+    elif event.instrument == "PBR":
+      self.orders.append(Order("PBR", -1, self.pbr + self.spread))
+      self.orders.append(Order("PETR3", 1, 0))
+
+  def push(self, event):
+    if self.first:
+      self.operate(event)
+      self.first = False
+    else:
+      for order in self.orders:
+        order.cancel(self.id, order.id)
+      self.operate(event)
+    return self.orders
+
+
+  def fill(self, id, price, quantity, status, event):
+    super().fill(id, price, quantity, status)
+    
+    if id in self.orders:
+      if quantity > 0: # vendido em self.pbr
+        Order(event.instrument, 1, 0)
+        Order(event.instrument, 1, 0)
+      else:
+        Order(event.instrument, -1, 0)
+        Order(event.instrument, -1, 0)
+
+print(evaluateHist(RSI(), {'IBOV':'^BVSP.csv'}))
+print(evaluateHist(RiosStrat(), {'IBOV':'^BVSP.csv'}))
+print(evaluateHist(PBRStrat(), {'PETR3':'PETR3.csv', 'USDBRL':'USDBRL.csv', 'PBR': 'PBR.csv'}))
```

Comment:
You should be using evaluateIntr(), not evaluateHist()

Reply:
True